### DISCUSSION TITLE
Non-threaded version

### DISCUSSION THREAD

**Discussion by steven-murray** [21cmfast/21cmFAST#217] [Ideas]
Given the difficulty of installing 21cmFAST on MacOS mostly due to openMP, I wonder if we could make a version that doesn't require openMP to install -- something like a compile-time define that just omits all the pragmas. 

I haven't thought through what new kind of complexity that might introduce, and it might not be worth it. I was hoping conda installs would fix most of those problems, but MacOS seems to be an endless pit of problems. 

What do people think?

**Comment by qyx268** [comment]
Good idea, considering most MacOS users are on laptop, openMP probably is not required anyway. When I was working on v2 for minihalos, I used #ifdef for all flags (although I didn't consider openMP), trying to make the code skipping all flag checks. It didn't end up being much quicker though. But we can surely do this for openMP.

**Comment by BradGreig** [comment]
I guess it's a hard one to know a priori. I would say there is limited benefit of only having a single threaded 21cmFAST, just because reasonable sized runs would take too long. However, on the flip side this is obviously preferable to not even being able to run it due to openMP issues.

I had a quick look at this compile-time behaviour to omit pragmas, it doesn't seem too difficult to implement. However, how that would translate to adding it to the existing infrastructure would be another matter entirely.

What is the issue with the conda install? I find going down the route of gcc is always preferable to clang. Do you have any ability to force that?

**Reply by steven-murray** [reply]
I think @qyx268 is right in saying that this is typically only a problem on MacOS, which is usually only used on a laptop, where no big runs are going to be run anyway.

I don't *think* this will be difficult to implement. The infrastructure I'm more worried about is how to have two versions of this in `conda-forge`. I think everything else should be simple.

As for `conda` -- it seems we are still receiving issues about installing with conda on MacOS. Perhaps they're easily solvable anyway, but I figured if we have this available, MacOS users can also install with simple clang.
